locked-asset/token-unstake/src/unbond_tokens.rs

```diff
@@ -20,41 +22,24 @@ pub trait UnbondTokensModule:
         let mut penalty_tokens = ManagedVec::<Self::Api, _>::new();
         let mut processed_count = 0;
 
-        self.unlocked_tokens_for_user(&caller)
+        let new_unlocked_tokens = self
+            .unlocked_tokens_for_user(&caller)
             .update(|user_entries| {
                 while !user_entries.is_empty() && processed_count < MAX_CLAIM_UNLOCKED_TOKENS {
                     let entry = user_entries.get(0);
                     if current_epoch < entry.unlock_epoch {
                         break;
                     }
 
-                    let locked_tokens = entry.locked_tokens;
-                    let unlocked_tokens = entry.unlocked_tokens;
-
-                    // we only burn the tokens that are not unlocked
-                    // the rest are sent back as penalty
-                    let locked_tokens_burn_amount = unlocked_tokens.amount.clone();
-                    self.send().esdt_local_burn(
-                        &locked_tokens.token_identifier,
-                        locked_tokens.token_nonce,
-                        &locked_tokens_burn_amount,
-                    );
-
-                    let penalty_amount = &locked_tokens.amount - &unlocked_tokens.amount;
-                    if penalty_amount > 0 {
-                        let penalty = EsdtTokenPayment::new(
-                            locked_tokens.token_identifier,
-                            locked_tokens.token_nonce,
-                            penalty_amount,
-                        );
-                        penalty_tokens.push(penalty);
-                    }
+                    self.handle_single_unbond_entry(&entry, &mut penalty_tokens);
```

Comment:
Keeping in mind that the previous version of token-unstake will go live in a few days, I don't think a new upgrade will be done any time soon. So I would avoid having any code refactor, as that will increase the difficulty in the release process, by bringing a new codehash. And anyways, it will not be deployed on mainnet.

Reply:
I really don't care about code hashes, and I refuse to keep bad code around just for that silly reason. There's literally nothing to test, it's the same code.